tests/integration/operator/operator_test.go

```diff
@@ -0,0 +1,273 @@
+								Should(ContainElement("istio-cni-node"), "CNI DaemonSet is not deployed; expected list to contain element")
+
+							Expect(kubectl.GetPodPhase(namespace, "k8s-app=istio-cni-node")).To(Equal("Running"), "CNI Daemon is not running; unexpected Phase")
+							Success("CNI DaemonSet is deployed in the namespace and Running")
+						} else {
+							Consistently(kubectl.GetDaemonSets).
+								WithArguments(namespace).WithTimeout(30*time.Second).
+								Should(BeEmpty(), "CNI DaemonSet is present; expected list to be empty")
+							Success("CNI DaemonSet is not deployed in the namespace because it's not OpenShift")
+						}
+					})
+
+					It("doesn't continuously reconcile the Istio CR", func() {
+						Eventually(kubectl.Logs).
+							WithArguments(controlPlaneNamespace, "app=istiod", 30*time.Second).
+							ShouldNot(ContainSubstring("Reconciliation done"), "Istio Operator is continuously reconciling")
+						Success("Istio Operator stopped reconciling")
+					})
+				})
+
+				When("the Istio CR is deleted", func() {
+					BeforeEach(func() {
+						Expect(kubectl.Delete(controlPlaneNamespace, "istio", istioName)).To(Succeed(), "Istio CR failed to be deleted")
+						Success("Istio CR deleted")
+					})
+
+					It("removes everything from the namespace", func() {
+						Eventually(kubectl.GetResourceList).
+							WithArguments(controlPlaneNamespace, "all").
+							Should(Equal(kubectl.EmptyResourceList), "Namespace should be empty")
+						Success("Namespace is empty")
+					})
+				})
+			})
+		}
+
+		AfterAll(func() {
+			if CurrentSpecReport().Failed() {
+				LogDebugInfo()
+			}
+			By("Cleaning up the namespace")
+			Expect(kubectl.DeleteNamespace(controlPlaneNamespace)).
+				To(Succeed(), "Namespace failed to be deleted")
+
+			Eventually(kubectl.CheckNamespaceExist).
+				WithArguments(controlPlaneNamespace).
+				Should(MatchError(kubectl.ErrNotFound), "Namespace should not exist")
+			Success("Cleanup done")
+		})
+	})
+
+	AfterAll(func() {
+		By("Cleaning up the operator")
+		Expect(helm.Uninstall("sail-operator", "--namespace "+namespace)).
+			To(Succeed(), "Operator failed to be deleted")
+		Expect(kubectl.DeleteCRDs(crds)).To(Succeed(), "CRDs failed to be deleted")
+		Success("Operator is deleted")
+	})
+})
+
+func LogDebugInfo() {
+	// General debugging information to help diagnose the failure
+	GinkgoWriter.Println("********* Failed specs while running: ", CurrentSpecReport().FailureLocation())
+	resource, err := kubectl.GetYAML(controlPlaneNamespace, "istio", istioName)
+	if err != nil {
+		GinkgoWriter.Println("Error getting Istio CR: ", err)
+	}
+	GinkgoWriter.Println("Istio CR: \n", resource)
+
+	output, err := kubectl.GetPods(controlPlaneNamespace, "-o wide")
+	if err != nil {
+		GinkgoWriter.Println("Error getting pods: ", err)
+	}
+	GinkgoWriter.Println("Pods in Istio CR namespace: \n", output)
+
+	logs, err := kubectl.Logs(namespace, "control-plane=sail-operator", 120*time.Second)
```

Comment:
We should actually remove the `control-plane` label from the operator. Ideally, here we would look up the logs from deployment (i.e. `deploy/sail-operator`) instead of using a label selector. However, for the time being, we should use the following selector: `app.kubernetes.io/instance=sail-operator`.